src/sku/molecule/services/src/use_cases/molecule_enable_project_use_case_impl.rs

```diff
@@ -0,0 +1,154 @@
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::sync::Arc;
+
+use internal_error::ResultIntoInternal;
+use kamu_accounts::LoggedAccount;
+use kamu_core::PushIngestDataUseCase;
+use kamu_core::auth::DatasetAction;
+use messaging_outbox::{Outbox, OutboxExt};
+use odf::metadata::OperationType;
+
+use crate::domain::*;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[dill::component]
+#[dill::interface(dyn MoleculeEnableProjectUseCase)]
+pub struct MoleculeEnableProjectUseCaseImpl {
+    molecule_dataset_service: Arc<dyn MoleculeDatasetService>,
+    push_ingest_use_case: Arc<dyn PushIngestDataUseCase>,
+    outbox: Arc<dyn Outbox>,
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[common_macros::method_names_consts]
+#[async_trait::async_trait]
+impl MoleculeEnableProjectUseCase for MoleculeEnableProjectUseCaseImpl {
+    #[tracing::instrument(
+        level = "info",
+        name = MoleculeEnableProjectUseCaseImpl_execute,
+        skip_all,
+        fields(?ipnft_uid)
+    )]
+    async fn execute(
+        &self,
+        molecule_subject: &LoggedAccount,
+        ipnft_uid: String,
+    ) -> Result<MoleculeProjectEntity, MoleculeEnableProjectError> {
+        use datafusion::prelude::*;
+
+        let (projects_dataset, df_opt) = self
+            .molecule_dataset_service
+            .get_projects_raw_ledger_data_frame(molecule_subject, DatasetAction::Write, false)
+            .await?;
+
+        let Some(ledger_df): Option<odf::utils::data::DataFrameExt> = df_opt else {
+            return Err(MoleculeEnableProjectError::ProjectNotFound(
+                ProjectNotFoundError {
+                    ipnft_uid: ipnft_uid.clone(),
+                },
+            ));
+        };
+
+        // Reconstruct active snapshot to confirm there are no conflicts
+        let snapshot_df = odf::utils::data::changelog::project(
+            ledger_df.clone(),
+            &["account_id".to_string()],
+            &odf::metadata::DatasetVocabulary::default(),
+        )
+        .int_err()?;
+
+        let df = ledger_df
+            .filter(col("ipnft_uid").eq(lit(&ipnft_uid)))
+            .int_err()?
+            .sort(vec![col("system_time").sort(false, false)])
+            .int_err()?;
+
+        let records: Vec<serde_json::Value> = df.collect_json_aos().await.int_err()?;
+        let record = records.into_iter().find(|record: &serde_json::Value| {
+            record
+                .get("op")
+                .and_then(serde_json::Value::as_u64)
+                .and_then(|value| u8::try_from(value).ok())
+                .and_then(|value| OperationType::try_from(value).ok())
+                .map(|op| matches!(op, OperationType::Append | OperationType::CorrectTo))
```

Comment:
If we are enabling the project, shouldn't we be looking for a previously retracted (disabled) project entry?